tests/CMakeLists.txt

```diff
@@ -124,34 +116,42 @@ message(STATUS "${msg}")
 
 # *DO* use json_test_set_test_options() above this line
 
-file(GLOB files src/unit-*.cpp)
-list(FILTER files EXCLUDE REGEX "src/unit-32bit.cpp")
+string(TOUPPER "${JSON_32bitTest}" json_32bit_test)
+if("${json_32bit_test}" STREQUAL AUTO)
+    # check if compiler is targeting 32bit by default
+    include(CheckTypeSize)
+    check_type_size("size_t" sizeof_size_t LANGUAGE CXX)
+    if(sizeof_size_t AND ${sizeof_size_t} EQUAL 4)
+        message(STATUS "Auto-enabling 32bit unit test.")
+        set(json_32bit_test ON)
+    else()
+        set(json_32bit_test OFF)
+    endif()
+endif()
+
+if("${json_32bit_test}" STREQUAL ONLY)
+    set(files src/unit-32bit.cpp)    
+else()
+    file(GLOB files src/unit-*.cpp)
+    if("${json_32bit_test}" STREQUAL OFF)
+        list(FILTER files EXCLUDE REGEX src/unit-32bit.cpp)
+    endif()
+endif()
 foreach(file ${files})
     json_test_add_test_for(${file} MAIN test_main CXX_STANDARDS ${test_cxx_standards} ${test_force})
 endforeach()
 
-if(JSON_32bitTest)
-    add_library(test_main32 OBJECT ${test_main_SOURCES})
-    target_compile_definitions(test_main32 ${test_main_COMPILE_DEFINITIONS})
-    target_compile_features(test_main32 ${test_main_COMPILE_FEATURES})
-    target_compile_options(test_main32 ${test_main_COMPILE_OPTIONS} -m32)
-    target_include_directories(test_main32 ${test_main_INCLUDE_DIRECTORIES})
-    target_link_libraries(test_main32 ${test_main_LINK_LIBRARIES})
-    target_link_options(test_main32 PUBLIC -m32)
-
-    json_test_add_test_for("src/unit-32bit.cpp" MAIN test_main32
-        CXX_STANDARDS ${test_cxx_standards} ${test_force})
+if(NOT "${json_32bit_test}" STREQUAL ONLY)
+    # test legacy comparison of discarded values
+    json_test_set_test_options(test-comparison_legacy
+        COMPILE_DEFINITIONS JSON_USE_LEGACY_DISCARDED_VALUE_COMPARISON=1
```

Comment:
Why does this depend on `json_32bit_test`?

Reply:
It depends on `JSON_32bitTest=ONLY`. No other tests should be built. Technically, that also applies to the CMake tests below.

Reply:
This feels over-engineered to me. Why do we need the flag in the first place?

Wouldn't it be sufficient to check, inside the coverage target, to add the 32 bit unit test with `-m32` flag?

Reply:
It's not possible to "just" build a single unit test with `-m32` because that also needs the shared code (test_main) built with `-m32`. That's what I wanted to avoid in this "simplified" version. Instead, we *only* build the 32bit test. But I do have another idea. Just add a `return()` to the `CMakeLists.txt` and skip everything else in the file. I'll check if that's possible.